pkg/controllers/pushsecret/pushsecret_controller_test.go

```diff
@@ -0,0 +1,442 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pushsecret
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
```

Comment:
I see we are not using Kubebuilder EnvTest here, and also using Ginkgo with a somewhat different approach when compared to other controllers, for unit tests. In our other controllers we don't do "pure" unit tests, to already check initial integration with kubernetes api.

I would like to understand the reasoning and see if we could get this standardized (Or changing other controllers if we decide that, or making this one follow same standard). 

cc @moolen @BooleanCat @gusfcarvalho